app/src/main/java/com/example/cinemapp/ui/main/DetailsFragment.kt

```diff
@@ -0,0 +1,159 @@
+package com.example.cinemapp.ui.main
+
+import android.graphics.Typeface
+import android.os.Bundle
+import android.text.SpannableString
+import android.text.Spanned
+import android.text.TextPaint
+import android.text.method.LinkMovementMethod
+import android.text.style.ClickableSpan
+import android.util.Log
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.core.math.MathUtils
+import androidx.core.view.size
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.LinearSmoothScroller
+import androidx.recyclerview.widget.PagerSnapHelper
+import com.example.cinemapp.R
+import com.example.cinemapp.databinding.FragmentDetailsBinding
+import com.example.cinemapp.util.observeFlowSafely
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+
+class DetailsFragment : Fragment() {
+
+    private var _binding: FragmentDetailsBinding? = null
+    private val binding get() = _binding!!
+    private val args: DetailsFragmentArgs by navArgs()
+    private val viewModel by viewModel<DetailsViewModel>()
+    private val castAdapter = CastAdapter()
+    private val mediaAdapter = MediaAdapter()
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentDetailsBinding.inflate(inflater, container, false)
+        binding.clContent.visibility = View.INVISIBLE
+        viewModel.getMovieDetails(args.movieId)
+        setupAdapter()
+        return binding.root
+    }
+
+    override fun onStart() {
+        super.onStart()
+        observeFlowSafely(viewModel.state) {
+            it.details?.let { details ->
+                setupView(details)
+                binding.clContent.visibility = View.VISIBLE
+                binding.cpiLoading.visibility = View.GONE
+            }
+            castAdapter.setCast(it.cast)
+            mediaAdapter.setMedia(it.media)
+        }
+    }
+
+    private fun setupView(details: MovieDetails) {
+        with(binding) {
+            tbTop.setNavigationIcon(R.drawable.vic_arrow_back)
```

Comment:
Toolbar is one of the only Views whose name should not be shortened, so please change the view ID form `tbTop` to `toolbar`